packages/back-end/src/controllers/stripe.ts

```diff
@@ -235,6 +236,29 @@ export async function postWebhook(req: Request, res: Response) {
         await updateSubscriptionInDb(subscription);
         break;
       }
+
+      case "customer.subscription.trial_will_end": {
+        const responseSubscription = event.data
+          .object as Stripe.Response<Stripe.Subscription>;
+        if (responseSubscription) {
+          const ret = await updateSubscriptionInDb(responseSubscription);
+          if (ret) {
+            const { organization, subscription, hasPaymentMethod } = ret;
+            const endDate = subscription.trial_end
+              ? new Date(subscription.trial_end * 1000)
+              : new Date(new Date().getTime() + 3 * 24 * 60 * 60 * 1000);
```

Comment:
If there's no trial_end for the subscription, something went wrong, and we should skip sending the email (and log an error).